src/sql/workbench/api/node/mainThreadModelViewDialog.ts

```diff
@@ -40,12 +40,19 @@ export class MainThreadModelViewDialog implements MainThreadModelViewDialogShape
 	}
 
 	public $openEditor(modelViewId: string, title: string, position?: vscode.ViewColumn): Thenable<void> {
-		let input = new ModelViewInput(title, modelViewId);
-		let editorOptions = Object.assign({
-			preserveFocus: true,
-			pinned: true
-		} as IEditorOptions);
-		return this._editorService.openEditor(input, editorOptions, position as any).then(() => undefined);
+		return new Promise<void>((resolve, reject) => {
+			let input = new ModelViewInput(title, modelViewId);
+			let editorOptions = {
+				preserveFocus: true,
+				pinned: true
+			};
+
+			this._editorService.openEditor(input, editorOptions, position as any).then(() => {
+				resolve();
+			}, error => {
+				reject();
```

Comment:
This should be `reject(error)` so that the error doesn't get lost.

In any case why the change from before? If your `then` callback doesn't have an error handler then that promise will just be rejected